c_glib/arrow-glib/decimal.cpp

```diff
@@ -0,0 +1,185 @@
+  std::shared_ptr<arrow::Decimal128> decimal128;
+} GArrowDecimal128Private;
+
+enum {
+  PROP_0,
+  PROP_DECIMAL128
+};
+
+G_DEFINE_TYPE_WITH_PRIVATE(GArrowDecimal128,
+                           garrow_decimal128,
+                           G_TYPE_OBJECT)
+
+#define GARROW_DECIMAL128_GET_PRIVATE(obj)                 \
+  (G_TYPE_INSTANCE_GET_PRIVATE((obj),                   \
+                               GARROW_TYPE_DECIMAL128,     \
+                               GArrowDecimal128Private))
+
+static void
+garrow_decimal128_finalize(GObject *object)
+{
+  auto priv = GARROW_DECIMAL128_GET_PRIVATE(object);
+
+  priv->decimal128 = nullptr;
+
+  G_OBJECT_CLASS(garrow_decimal128_parent_class)->finalize(object);
+}
+
+static void
+garrow_decimal128_set_property(GObject *object,
+                        guint prop_id,
+                        const GValue *value,
+                        GParamSpec *pspec)
+{
+  auto priv = GARROW_DECIMAL128_GET_PRIVATE(object);
+
+  switch (prop_id) {
+  case PROP_DECIMAL128:
+    priv->decimal128 =
+      *static_cast<std::shared_ptr<arrow::Decimal128> *>(g_value_get_pointer(value));
+    break;
+  default:
+    G_OBJECT_WARN_INVALID_PROPERTY_ID(object, prop_id, pspec);
+    break;
+  }
+}
+
+static void
+garrow_decimal128_init(GArrowDecimal128 *object)
+{
+}
+
+static void
+garrow_decimal128_class_init(GArrowDecimal128Class *klass)
+{
+  GParamSpec *spec;
+
+  auto gobject_class = G_OBJECT_CLASS(klass);
+
+  gobject_class->finalize     = garrow_decimal128_finalize;
+  gobject_class->set_property = garrow_decimal128_set_property;
+
+  spec = g_param_spec_pointer("decimal128",
+                              "Decimal128",
+                              "The raw std::shared<arrow::Decimal128> *",
+                              static_cast<GParamFlags>(G_PARAM_WRITABLE |
+                                                       G_PARAM_CONSTRUCT_ONLY));
+  g_object_class_install_property(gobject_class, PROP_DECIMAL128, spec);
+}
+
+/**
+ * garrow_decimal128_new_string:
+ * @data: The data of the decimal.
+ *
+ * Returns: A newly created #GArrowDecimal128.
+ *
+ * Since: 1.0.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_new_string(const gchar *data)
+{
+  auto arrow_decimal = std::make_shared<arrow::Decimal128>(data);
+  return garrow_decimal128_new_raw(&arrow_decimal);
+}
+
+/**
+ * garrow_decimal128_new_integer:
+ * @data: The data of the decimal.
+ *
+ * Returns: A newly created #GArrowDecimal128.
+ *
+ * Since: 1.0.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_new_integer(const gint64 data)
+{
+  auto arrow_decimal = std::make_shared<arrow::Decimal128>(data);
+  return garrow_decimal128_new_raw(&arrow_decimal);
+}
+
+/**
+ * garrow_decimal128_to_string:
+ * @decimal: A #GArrowDecimal128.
+ * @scale: The scale of the decimal.
+ *
+ * Returns: The string representation of the decimal.
+ *
+ * Since: 1.0.0
+ */
+const gchar *
+garrow_decimal128_to_string(GArrowDecimal128 *decimal, gint32 scale)
```

Comment:
How about removing `gint32 scale` argument from this function?
Because `*_to_string()` is a general name for a function that doesn't accept no extra argument in GLib context.
For example:

  * [`g_quark_to_string()`](https://developer.gnome.org/glib/stable/glib-Quarks.html#g-quark-to-string)
  * [`gtk_css_provider_to_string()`](https://developer.gnome.org/gtk3/stable/GtkCssProvider.html#gtk-css-provider-to-string)
 * [`gtk_tree_path_to_string()`](https://developer.gnome.org/gtk3/stable/GtkTreeModel.html#gtk-tree-path-to-string)

If we remove `gint32 scale` argument, we can use `arrow::Decimal128::ToIntegerString()` internally.

Reply:
How about `garrow_decimal128_to_string_scale(GArrowDecimal128 *decimal, gint32 scale)` as the new name?